include/nbl/video/ILogicalDevice.h

```diff
@@ -177,6 +173,12 @@ class NBL_API2 ILogicalDevice : public core::IReferenceCounted, public IDeviceMe
                     for (const auto& queue : *m_queues)
                         queue->cullResources(info.semaphore);
                 }
+                if (waitStatus == retval_t::DEVICE_LOST)
+                {
+                    m_logger.log("Device lost [%s - %s:%d]", system::ILogger::ELL_ERROR, __FUNCTION__, __FILE__, __LINE__);
```

Comment:
@devshgraphicsprogramming What do you think about a a `LOG_ERROR` macro.
It allows us to change to format of how we log in the future and add `GIT_COMMIT_HASH`

Reply:
I say macro because __FUNCTION__, __LINE__ need to work :D

Reply:
yes but such a macro should not be in a header shared between multiple cpp, just each of the CPPs

alternatively it can be defined as a so-called X-macro (a short header you include, but need to define some defines first for the macros inside to work, e.g. `m_logger` for the logger object)